src/app/query/fees/fees.hooks.ts

```diff
@@ -1,11 +1,81 @@
+import { useMemo } from 'react';
+
+import { useAnalytics } from '@app/common/hooks/analytics/use-analytics';
+import { FeeEstimate, TransactionFeeEstimation } from '@shared/models/fees-types';
+import {
+  useConfigFeeEstimationsMaxEnabled,
+  useConfigFeeEstimationsMaxValues,
+  useConfigFeeEstimationsMinEnabled,
+  useConfigFeeEstimationsMinValues,
+} from '@app/query/hiro-config/hiro-config.query';
+import { useGetTransactionFeeEstimationQuery } from '@app/query/fees/fees.query';
+
+import { getDefaultSimulatedFeeEstimations, getFeeEstimationsWithCappedValues } from './fees.utils';
 import { UseQueryResult } from 'react-query';
 
-import { TransactionFeeEstimation } from '@shared/models/fees-types';
-import { useGetFeeEstimations } from '@app/query/fees/fees.query';
+const defaultFeesMaxValues = [500000, 750000, 2000000];
+const defaultFeesMinValues = [2500, 3000, 3500];
 
-export function useFeeEstimationsQuery(
-  transactionPayload: string,
-  estimatedLen: number | null
+const defaultFeeEstimations: FeeEstimate[] = [
+  { fee: defaultFeesMinValues[0], fee_rate: 0 },
+  { fee: defaultFeesMinValues[1], fee_rate: 0 },
+  { fee: defaultFeesMinValues[2], fee_rate: 0 },
+];
+
+function useTransactionFeeEstimation(
+  estimatedLen: number | null,
+  transactionPayload: string
 ): UseQueryResult<TransactionFeeEstimation, Error> {
-  return useGetFeeEstimations(transactionPayload, estimatedLen);
+  return useGetTransactionFeeEstimationQuery(estimatedLen, transactionPayload);
+}
+
+function useFeeEstimationsMaxValues() {
+  const configFeeEstimationsMaxEnabled = useConfigFeeEstimationsMaxEnabled();
+  const configFeeEstimationsMaxValues = useConfigFeeEstimationsMaxValues();
+
+  if (configFeeEstimationsMaxEnabled === false) return;
+  return configFeeEstimationsMaxValues || defaultFeesMaxValues;
+}
+
+function useFeeEstimationsMinValues() {
+  const configFeeEstimationsMinEnabled = useConfigFeeEstimationsMinEnabled();
+  const configFeeEstimationsMinValues = useConfigFeeEstimationsMinValues();
+
+  if (configFeeEstimationsMinEnabled === false) return;
+  return configFeeEstimationsMinValues || defaultFeesMinValues;
+}
+
+export function useFeeEstimations(txByteLength: number | null, txPayload: string) {
+  const analytics = useAnalytics();
+  const feeEstimationsMaxValues = useFeeEstimationsMaxValues();
+  const feeEstimationsMinValues = useFeeEstimationsMinValues();
+  const result = useTransactionFeeEstimation(txByteLength, txPayload);
+  const { data: txFeeEstimation, isError, isLoading } = result;
+
+  return useMemo(() => {
+    const feeEstimations = txFeeEstimation?.estimations;
+
+    if (!isLoading && isError) {
+      void analytics.track('use_default_fee_estimations');
+      return defaultFeeEstimations;
+    }
+    if (!isLoading && (!!txFeeEstimation?.error || !feeEstimations?.length) && txByteLength) {
+      void analytics.track('use_default_simulated_fee_estimations');
```

Comment:
I don't think we want to trigger an event in a memo, this could fire at a render pace if this condition is true.

Reply:
Good point, I will remove the memo.

Reply:
Removing the memo alone will just trigger the events during "render time" instead of "memo reevaluation time". Is that what you meant?

Perhaps we should log these at the time at which the user selects the fee they want, or at the time they broadcast?

Reply:
Yep, intend to move them ...just trying to think thru how we will know which fees set are being used: default, simulated, or api/capped.

Reply:
Maybe this API return something like the nonce calc?

```tsx
interface Resp { 
  estimates: FeeEstimate[];
  calculation: 'default' | 'default-simulated' | 'fees-capped' 
}
```